05_iigo/index.tex

```diff
@@ -583,4 +578,254 @@ \subsubsection{Server-side}
+\begin{rule_IIGO}
+    "If you are expected to pay $X$ amount of taxation, the amount of tax you pay must be $X$."
+\label{rule:basicTaxRule}
+\end{rule_IIGO}
+\par
+Notice that rule \ref{rule:basicTaxRule} is a rather simple one. However, it allows us to form a basis for how to represent rules as matrices.
+If our goal is simply to ensure that an agent pays the expected amount of tax, a basic mathematical operation we can perform is to subtract the actual tax paid from the expected.
+If we obtain $0$ from this subtraction, we can conclude that this agent has adhered to this rule. \\
+Formally, let $x$ and $y$ be the expected and the actual paid amount of tax, respectively. To confirm adherence to the rule, we want
+\begin{equation}
+    y - x = 0 . 
+    \label{equation:basicMaths}
+\end{equation}
+
+To turn \ref{equation:basicMaths}  into a matrix calculation, we simply have to introduce some trivial coefficients.
+\begin{equation}
+    -1*x +
+       1*y = 0
+       \label{equation:coefLinear}
+\end{equation}
+      
+Now we can write \ref{equation:coefLinear} as a matrix calculation.
+\begin{equation}
+    \begin{bmatrix}
+        -1 & 1
+    \end{bmatrix}
+    \begin{bmatrix}
+        x \\
+        y
+    \end{bmatrix}   
+    = 0  
+    \label{equation:firstMatrix}
+\end{equation}
+
+We have now encoded this simple rule given as an example in terms of a matrix product. An agent can input their versions of $x$ and $y$, and perform the calculation.
+The agents can then check if their result of \ref{equation:firstMatrix} is equal to $0$. Let us examine a slightly more complicated rule. \\
+\begin{rule_IIGO}
+    "If you are expected to pay $x$ amount of tax, the amount of tax you pay must be at least $x$."
+\end{rule_IIGO}
+
+We can rewrite the matrix equation as an inequality.
+\begin{equation}
+    \begin{bmatrix}
+        -1 & 1
+    \end{bmatrix}
+    \begin{bmatrix}
+        x \\
+        y
+    \end{bmatrix}   
+    \geq 0
+    \label{equation:basicInequality}  
+\end{equation}
+The addition of the "at least" (i.e. $\geq$) in \ref{equation:basicInequality} adds a little bit of complexity to our previously introduced matrix scheme. For this instance, we may not be able to simply check for equality and but instead, need to check for a $\geq$.
+We made a decision that we would ensure all of our calculations would be compared to $0$. This allowed us to reduce the complexity of our calculation algorithms.
+However, with this simplification we encounter a problem with a rule like \ref{rule:taxWithOffset}.
+\begin{rule_IIGO}
+    "If you are expected to pay $x$ amount of tax, the amount of tax you pay must be at least $x + 5$."
+    \label{rule:taxWithOffset}
+\end{rule_IIGO}
+In this case, it would be impossible to capture this rule using the previously suggested matrix scheme. Therefore, to capture such cases of rules, we introduce a \emph{constant} to the input list, as below in \ref{equation:inequalityWithOffset}.
```

Comment:
```suggestion
In this case, it would be impossible to capture this rule using the previously suggested matrix scheme. Therefore, to capture such cases of rules, we introduce a \emph{constant} to the input list, as below in Equation~\eqref{equation:inequalityWithOffset}.
```